modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.web.internal.notifications;
+
+import com.liferay.asset.display.page.portlet.AssetDisplayPageFriendlyURLProvider;
+import com.liferay.info.item.InfoItemReference;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.UserNotificationEvent;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * @author Thalles Montenegro
+ */
+public class ObjectUserNotificationsHandlerTest {
+
+	@ClassRule
+	public static LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		_objectUserNotificationsHandler = new ObjectUserNotificationsHandler(
+			_assetDisplayPageFriendlyURLProvider, _objectDefinition);
+
+		_portletURLStep = new PortletURLBuilder.PortletURLStep(
+			Mockito.mock(PortletURL.class));
+
+		_portletURL = Mockito.mock(PortletURL.class);
+	}
+
+	@Test
+	public void testGetLinkWhenNotificationHasFriendlyURL() throws Exception {
+		Mockito.when(
+			_userNotificationEvent.getPayload()
+		).thenReturn(
+			""
+		);
+
+		Mockito.when(
+			_serviceContext.getThemeDisplay()
+		).thenReturn(
+			Mockito.mock(ThemeDisplay.class)
+		);
+
+		Mockito.when(
+			_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
+				Mockito.any(InfoItemReference.class),
+				Mockito.any(ThemeDisplay.class))
+		).thenReturn(
+			"friendlyURL"
+		);
+
+		Assert.assertEquals(
+			"friendlyURL",
+			_objectUserNotificationsHandler.getLink(
+				_userNotificationEvent, _serviceContext));
+	}
```

Comment:
Hey @thallescezar please apply my suggestion below:

```java
	public void testGetLinkWhenObjectEntryHasDisplayPageFriendlyURL()
		throws Exception {

		Mockito.when(
			_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
				Mockito.any(InfoItemReference.class),
				Mockito.any(ThemeDisplay.class))
		).thenReturn(
			"http://localhost:8080/web/l/54321"
		);

		Assert.assertEquals(
			"http://localhost:8080/web/l/54321",
			_objectUserNotificationsHandler.getLink(
				_userNotificationEvent, _serviceContext));
	}
```

Thanks for covering this case as well!